teamcity: failed tests on master: testrace/TestDebugLogSpyBrokenConnection

The following tests appear to have failed:

[#344713](https://teamcity.cockroachdb.com/viewLog.html?buildId=344713):

```
--- FAIL: testrace/TestDebugLogSpyBrokenConnection (0.000s)
Test ended in panic.

------- Stdout: -------
panic: test timed out after 15m0s

goroutine 29958 [running]:
testing.startAlarm.func1()
	/usr/local/go/src/testing/testing.go:1023 +0x139
created by time.goFunc
	/usr/local/go/src/time/sleep.go:170 +0x52

goroutine 1 [chan receive, 13 minutes]:
testing.(*T).Run(0xc4204b05b0, 0x22f6ce3, 0x1f, 0x239ea50, 0xc4205c1b01)
	/usr/local/go/src/testing/testing.go:698 +0x582
testing.runTests.func1(0xc4204b05b0)
	/usr/local/go/src/testing/testing.go:882 +0xab
testing.tRunner(0xc4204b05b0, 0xc4205c1d10)
	/usr/local/go/src/testing/testing.go:657 +0x108
testing.runTests(0xc4201f0660, 0x31b3ce0, 0x53, 0x53, 0x13)
	/usr/local/go/src/testing/testing.go:888 +0x4e1
testing.(*M).Run(0xc420975f28, 0x405bfa0)
	/usr/local/go/src/testing/testing.go:822 +0x1c4
github.com/cockroachdb/cockroach/pkg/server_test.TestMain(0xc4205c1f28)
	/go/src/github.com/cockroachdb/cockroach/pkg/server/main_test.go:32 +0x185
main.main()
	github.com/cockroachdb/cockroach/pkg/server/_test/_testmain.go:206 +0x210

goroutine 17 [syscall, 15 minutes, locked to thread]:
runtime.goexit()
	/usr/local/go/src/runtime/asm_amd64.s:2197 +0x1

goroutine 6 [chan receive]:
github.com/cockroachdb/cockroach/pkg/util/log.(*loggingT).flushDaemon(0x3699f40)
	/go/src/github.com/cockroachdb/cockroach/pkg/util/log/clog.go:1004 +0x85
created by github.com/cockroachdb/cockroach/pkg/util/log.init.1
	/go/src/github.com/cockroachdb/cockroach/pkg/util/log/clog.go:580 +0x13a

goroutine 7 [syscall, 15 minutes]:
os/signal.signal_recv(0x704761)
	/usr/local/go/src/runtime/sigqueue.go:116 +0x104
os/signal.loop()
	/usr/local/go/src/os/signal/signal_unix.go:22 +0x30
created by os/signal.init.1
	/usr/local/go/src/os/signal/signal_unix.go:28 +0x4f

goroutine 19 [select, 13 minutes, locked to thread]:
runtime.gopark(0x23a2090, 0x0, 0x22d2c4c, 0x6, 0x18, 0x2)
	/usr/local/go/src/runtime/proc.go:271 +0xfd
runtime.selectgoImpl(0xc420af0f50, 0xc420af0f38, 0x0)
	/usr/local/go/src/runtime/select.go:423 +0x138a
runtime.selectgo(0xc420af0f50)
	/usr/local/go/src/runtime/select.go:238 +0x1c
runtime.ensureSigM.func1()
	/usr/local/go/src/runtime/signal_unix.go:434 +0x1ea
runtime.goexit()
	/usr/local/go/src/runtime/asm_amd64.s:2197 +0x1

goroutine 29982 [select]:
github.com/cockroachdb/cockroach/pkg/server.(*logSpy).run(0xc420977ef8, 0x7f5e7953f0c8, 0xc4205ba1e0, 0x31bd7c0, 0xc42046c080, 0x1001, 0x34630b8a000, 0x0, 0x0, 0x0)
	/go/src/github.com/cockroachdb/cockroach/pkg/server/debug_logspy.go:214 +0x7ef
github.com/cockroachdb/cockroach/pkg/server.TestDebugLogSpyBrokenConnection.func3(0xc420977ef8, 0x7f5e7c6cd928, 0xc420014e28, 0xc42046c080, 0xc420977f18, 0xc420d881a0)
	/go/src/github.com/cockroachdb/cockroach/pkg/server/debug_logspy_test.go:331 +0x9d
github.com/cockroachdb/cockroach/pkg/server.TestDebugLogSpyBrokenConnection(0xc420d881a0)
	/go/src/github.com/cockroachdb/cockroach/pkg/server/debug_logspy_test.go:334 +0x314
testing.tRunner(0xc420d881a0, 0x239ea50)
	/usr/local/go/src/testing/testing.go:657 +0x108
created by testing.(*T).Run
	/usr/local/go/src/testing/testing.go:697 +0x544
?   	github.com/cockroachdb/cockroach/pkg/server/diagnosticspb	[no test files]
?   	github.com/cockroachdb/cockroach/pkg/server/serverpb	[no test files]

```
Please assign, take a look and update the issue accordingly.

--- a/pkg/server/debug_logspy_test.go
+++ b/pkg/server/debug_logspy_test.go
@@ -327,7 +327,7 @@ func TestDebugLogSpyBrokenConnection(t *testing.T) {
 		func() {
 			if err := spy.run(ctx, w, logSpyOptions{
 				Duration: durationAsString(time.Hour),
-				Count:    logSpyChanCap + 1,
+				Count:    logSpyChanCap - 1, // will definitely see that many entries
 			}); !testutils.IsError(err, test.failStr) {
 				t.Fatal(err)
 			}

